src/vs/workbench/services/progress/browser/progressService.ts

```diff
@@ -10,16 +10,39 @@ import { IViewletService } from 'vs/workbench/services/viewlet/browser/viewlet';
 import { IPanelService } from 'vs/workbench/services/panel/common/panelService';
 import { IProgressService, IProgressRunner } from 'vs/platform/progress/common/progress';
 
-interface ProgressState {
-	infinite?: boolean;
-	total?: number;
-	worked?: number;
-	done?: boolean;
-	whilePromise?: Promise<any>;
-	whileStart?: number;
-	whileDelay?: number;
+
+const NoneProgressState = new class { readonly type = 'none'; };
```

Comment:
@mjbvz I always use const enums for such cases, I prefer the lower memory overhead of using integers compared to strings and that I can find all places where the type is checked with a simple search of the enum name. Plus, const enums get inlined (as ints) in the JS output even across different modules. An enum has the added benefit of immediately communicating that this is a closed set of types. i.e.

```ts
const enum ProgressType {
 None,
 Done,
 Infinite,
 While,
 Work
}

const NoneProgressState = new class { readonly type = ProgressType.None; };

// .. etc
```

Reply:
That makes sense. I've updated this PR to use a const enum and move all the relevant types into a namespace